index.js

```diff
@@ -84,8 +84,8 @@ function parse(str, options) {
  */
 
 function serialize(name, val, options) {
-  var opt = options || {};
-  var enc = opt.encode || encode;
+  var opt = typeof options === 'object' ? options : {};
+  var enc = typeof opt.encode === 'function' ? opt.encode : encode;
```

Comment:
Modified it to apply those changes (only falls through to the default if the "encode" property hasn't been defined for the `options` object.